client/services/utils/local_settings.js

```diff
@@ -0,0 +1,65 @@
+import EventEmitter from 'event-emitter-es6'
+
+const STORAGE_KEY = 'localsettings'
+const DATA_TYPE_KEY = 'dataType'
+const VALUE_KEY = 'value'
+const NAME_KEY = 'name'
+class LocalSettings extends EventEmitter {
+  constructor ($window, $rootScope) {
+    super()
+    Object.assign(this, { $window, $rootScope })
+    const settingsFromSession = this.getFromLocalStorage()
+    if (settingsFromSession !== undefined) {
+      this.settings = settingsFromSession
+    } else {
+      this.settings = { }
+    }
+  }
+
+  // example:
+  /**
+   * example input: settingKey: 'Scoresheel-Autoscroll' settingValueTypePair: {value: true, dataType: 'boolean'}
+   * @param {string} settingsKey
+   * @param {{value:*,dataType:string}} settingValueTypePair
+   */
+  update (settingsKey, settingValueTypePair) {
+    this.settings[settingsKey] = settingValueTypePair
+    this.$rootScope.$watch(() => this.settings[settingsKey][VALUE_KEY], (newValue, oldValue) => {
```

Comment:
You don't need this `$watch` since this service is the only one updating these values. Remove it and remove `$rootScope` from this service